search/includes/classes/queue/class-cron.php

```diff
@@ -114,9 +125,47 @@ public function process_jobs( $job_ids ) {
 		$this->queue->process_jobs( $jobs );
 	}
 
+	/**
+	 * Given a term taxonomy id, queue all posts for reindexing that match it
+	 *
+	 * @param {int} $term_taxonomy_id The term taxonomy id you want to index
+	 */
+	public function queue_posts_for_term_taxonomy_id( $term_taxonomy_id ) {
+		$args = array(
+			'posts_per_page' => self::TERM_UPDATE_BATCH_SIZE,
+			'paged' => 1,
+			'tax_query' => array(
+				array(
+					'field' => 'term_taxonomy_id',
+					'terms' => $term_taxonomy_id,
+				),
+			),
+		);
+
+		$posts = new \WP_Query( $args );
+
+		if ( ! $posts->have_posts() ) {
+			return;
+		}
+
+		while ( $args['paged'] <= $posts->max_num_pages ) {
+			while ( $posts->have_posts() ) {
```

Comment:
`$posts->posts` will also just be a plain array here, so we can `foreach()` it like normal, which is a little cleaner.